Checkout/Checkout/Source/Validation/Validators/CardNumberValidator.swift

```diff
@@ -32,9 +33,32 @@ final class CardNumberValidator: CardNumberValidating {
       return .success(cardScheme)
     case .none:
 
-      let cardScheme = cardTypeMatch(cardNumber, \.eagerCardNumberRegex) ?? .unknown
-      return .success(cardScheme)
+      return .success(.unknown)
+    }
+  }
+
+  func eagerValidate(cardNumber: String) -> Result<Card.Scheme, ValidationError.EagerCardNumber> {
+    let cardNumber = cardNumber.filter { !$0.isWhitespace }
+
+    guard validateDigitsOnly(in: cardNumber) else {
+      return .failure(.cardNumber(.invalidCharacters))
     }
+
+    switch cardTypeMatch(cardNumber, \.eagerCardNumberRegex) {
+    case .some(let cardScheme):
+      return withinCardLengthLimit(cardNumber, for: cardScheme) ? .success(cardScheme) : .failure(.tooLong)
+    case .none:
+      return shouldHaveFoundScheme(cardNumber) ? .failure(.invalidScheme) : .success(.unknown)
+    }
+  }
+
+  private func withinCardLengthLimit(_ cardNumber: String, for scheme: Card.Scheme) -> Bool {
+    let maxCardLength = scheme.maxCardLength ?? Card.Scheme.maxCardLengthAllSchemes
+    return cardNumber.count <= maxCardLength
```

Comment:
This is a follow-up comment on the previous related comment. 
```suggestion
    cardNumber.count <= scheme.maxCardLength
```